packages/wifi_iot/android/src/main/java/com/alternadom/wifiiot/WifiIotPlugin.java

```diff
@@ -1198,6 +1232,21 @@ public void run() {
         // set ssid
         builder.setSsid(ssid);
         builder.setIsHiddenSsid(isHidden != null ? isHidden : false);
+        if (bssid != null) {
+          try {
+            final MacAddress macAddress = MacAddress.fromString(bssid);
+            builder.setBssid(macAddress);
+          } catch (IllegalArgumentException invalidRepresentation) {
+            handler.post(
+                new Runnable() {
+                  @Override
+                  public void run() {
+                    poResult.error("Error", "Invalid BSSID representation", "");
+                  }
+                });
+            return;
+          }
+        }
```

Comment:
Can we avoid this duplicate code for parsing bssid ?

Reply:
Done. We don't gain much, but it's cleaner.